libmscore/instrument.cpp

```diff
@@ -750,6 +769,72 @@ void Channel::read(XmlReader& e, Part* part)
       updateInitList();
       }
 
+//---------------------------------------------------------
+//   switchExpressive
+//    Switches channel from non-expressive to expressive patch or vice versa
+//    This works only with MuseScore General soundfont
+//---------------------------------------------------------
+
+void Channel::switchExpressive(MasterSynthesizer* m, bool expressive)
+      {
+      if (_userBankController)
+            return;
+
+      // Work out where the new expressive patch will be
+      // All expressive instruments are +1 bank higher than the
+      // normal counterparts, except on bank 0, where they are placed on bank 17
+      // and on bank 8, which uses bank 18 instead.
+      int searchBankNum;
+      if (expressive) {
+            int relativeBank = bank() % 129;
+            int newBankNum;
+            if (relativeBank == 0)
+                  newBankNum = 17;
+            else if (relativeBank == 8)
+                  newBankNum = 18;
+            else
+                  newBankNum = relativeBank + 1;
+
+            searchBankNum = (bank()/129)*129 + newBankNum;
```

Comment:
Redundant math?

Reply:
Maybe add a comment to clear that confusion?

Reply:
Fixed.

Reply:
Not a necessary change. But an alternative that avoids integer division:
`searchBankNum = bank() - (bank() % 129) + newBankNum;`

Reply:
Yes, that would work too. I'm going to leave it as is, if that's alright?